packages/flutter/lib/src/rendering/table.dart

```diff
@@ -420,6 +496,11 @@ class RenderTable extends RenderBox {
   // _children.length must be rows * columns
   List<RenderBox> _children = const <RenderBox>[];
 
+  /// The number of vertical alignment lines in this table.
+  ///
+  /// Changing the number of columns will cause the cells in the table to be
+  /// rearranged into the specified number of columns, which is an expensive
+  /// operation.
```

Comment:
It doesn't rearrange the cells. It "just" chops off any cells that were in columns that no longer exist.

It is still expensive though.